io/directory.go

```diff
@@ -38,6 +39,10 @@ type Directory interface {
 	// ForEachLink applies the given function to Links in the directory.
 	ForEachLink(context.Context, func(*ipld.Link) error) error
 
+	// EnumLinksAsync returns a channel which will receive Links in the directory
+	// as they are enumerated, where order is not gauranteed
+	EnumLinksAsync(context.Context) (<-chan format.LinkResult, error)
```

Comment:
nit (although I personally consider it kind of important): I'd prefer to avoid the *link* terminology which relates to the DAG layer and use other terms like directory *entry* (e.g., EnumEntries, I don't even consider the `Async` suffix that relevant but we can keep it), which is, granted, implemented through a DAG link but it's a term that can be more easily associated to the UnixFS/MFS layers. The main difference being that a HAMT directory actually has many internal links (because it's composed of potentially several nodes) but those links won't be returned here, just the links that cross this directory sub-graph boundary to other UnixFS sub-graphs (representing other files and dirs).

This is actually not related to this particular PR since it's something inherited from the intricate naming system the code historically had but since it will be expanding the UnixFS public API we can start to (slowly) shift away from that.

Reply:
Yea I think it's reasonable to change this in the future but for the moment the Link terminology is used heavily in the directory interface.